starters/next-drupal-starter/pages/articles/[...slug].js

```diff
@@ -23,50 +25,33 @@ export default function ArticleTemplate({ title, body, imgSrc, hrefLang }) {
 }
 
 export async function getStaticPaths(context) {
-  // TODO - locale increases the complexity enough here that creating a usePaths
-  // hook would be a good idea.
-  // Get paths for each locale.
-  const pathsByLocale = context.locales.map(async (locale) => {
-    const store = getCurrentLocaleStore(locale, globalDrupalStateAuthStores);
-
-    const articles = await store.getObject({
-      objectName: "node--article",
-      query: `
-          {
-            id
-            path {
-              alias
-            }
-          }
-        `,
-    });
-    return articles.map((article) => {
-      // matches everything after /articles/
-      const match = article.path.alias.match(/^\/articles\/(.*)$/);
-      const slug = match[1];
-
-      return { params: { slug: [slug] }, locale: locale };
-    });
-  });
-
-  // Resolve all promises returned as part of pathsByLocale.
-  const paths = await Promise.all(pathsByLocale).then((values) => {
-    // Flatten the array of arrays into a single array.
-    return [].concat(...values);
-  });
+  try {
+    const paths = await getPaths(
+      context,
+      globalDrupalStateStores,
+      "node--article",
+      "slug",
+      "articles"
+    );
 
-  return {
-    paths,
-    fallback: false,
-  };
+    return {
+      paths,
+      fallback: false,
+    };
+  } catch (error) {
+    console.error("Failed to fetch paths for articles:", error);
+  }
 }
 
 export async function getStaticProps(context) {
   const { locales, locale } = context;
   const multiLanguage = isMultiLanguage(locales);
   const lang = context.preview ? context.previewData.previewLang : locale;
 
-  const store = getCurrentLocaleStore(lang, globalDrupalStateAuthStores);
+  const store = getCurrentLocaleStore(
+    lang,
+    context.preview ? globalDrupalStateAuthStores : globalDrupalStateStores
```

Comment:
My concern here is that there will be cases where people will want to always use the authenticated store. I'm inclined to leave this for now though - it will be an improvement in most cases and is easy to change.

Reply:
I could see that. Hoping this* will all be smoothed out when we make the necessary enhancements to DrupalState.

*as in stores per language and auth/non-auth stores

Reply:
Yup was also thinking the same thing.